src/main/java/frc/robot/utils/diag/DiagAprilTags.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.utils.diag;
+
+import edu.wpi.first.networktables.DoubleArraySubscriber;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+/**
+ * A diagnostics class for digital encoder. The diagnostics will turn green once the encoder has traveled at least a given
+ * distance from its initial position (measured at initialization or after a reset)
+ */
+public class DiagAprilTags implements Diagnosable {
+    private DoubleArraySubscriber subscriber;
+    private String title;
+    private String name;
+    private GenericEntry networkTableEntry;
+
+    /**
+     * Constructor
+     * @param name            - the name of the unit. Will be used on the Shuffleboard
+     */
+    public DiagAprilTags(String title, String name) {
+        this.title = title;
+        this.name = name;
+        reset();
```

Comment:
Just noticed that `reset()` is being called in most of the diagnosable's constructors. Instead, it might make sense to reset diagnostics in `robotInit()`, so that these constructor resets aren't needed